packages/playwright-core/src/server/injected/injectedScript.ts

```diff
@@ -1015,6 +1015,31 @@ export class InjectedScript {
       }
     }
 
+    // Multi-Select/Combobox
+    {
+      if (expression === 'to.have.value' && options.expectedText?.length && options.expectedText.length >= 2) {
+        if (element.nodeName !== 'SELECT' || !(element as HTMLSelectElement).multiple)
+          throw this.createStacklessError('Not a select element with a multiple attribute');
+
+        const received = [...(element as HTMLSelectElement).selectedOptions].map(o => o.value);
+        if (received.length !== options.expectedText.length)
+          return { received, matches: false };
+
+        let i = 0;
+        const matchers = options.expectedText.map(e => new ExpectedTextMatcher(e));
+        let allMatchesFound = true;
+        for (const matcher of matchers) {
+          while (i < received.length && !matcher.matches(received[i]))
```

Comment:
This seems to implement `toContain` logic, but I'd expect the exact match logic instead.